labellines/core.py

```diff
@@ -30,17 +31,19 @@ def labelLine(
     ----------
     line : matplotlib.lines.Line
        The line holding the label
-    x : number
+    val : number
        The location in data unit of the label
+    axis : "x" | "y"
+        Reference axis for `val`.
```

Comment:
I am overall happy with the `val/axis` combination. Two notes though
- [ ] we need to keep the old (`yoffset`, `yoffset_logspace`) for backward compatibility
- [ ] add a note in the docstring below to explain how to use the axis kwa?